e2e/deployer.go

```diff
@@ -0,0 +1,44 @@
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// deployer interface modify the templates on demand and also help
+// to create or delete templates.
+type deployer interface {
+	replaceNamespaceInTemplate(string) (string, error)
+	runAction(string, string, string) error
+}
+
+type deploy struct{}
```

Comment:
There is a lot of duplication in the deployer for cephfs and rbd, it probably makes sense to combine the two into a single `type deployer`.

When creating the `deployer` instance, pass it the list of templates, maybe with an additional attribute, like:

```go

type deploymentTemplateStage string

var (
    deploymentTemplateStageInit = deploymentTemplateStage("init") // for driver object
    deploymentTemplateStageConfig = deploymentTemplateStage("config")
    deploymentTemplateStageProvisioner = deploymentTemplateStage("provisioner")
    ...
)

type deploymentTemplate struct {
    filename string
    stage deploymentTemplateStage
}

type deployer struct {
    templates []deploymentTemplate
}

cephfsDeployer := deployer{
    templates: []deploymentTemplate{
        {pathToDriverTemplate, deploymentTemplateStageInit},
        {pathToConfigTemplate, deploymentTemplateStageConfig},
        {pathToMoreConfigTemplate, deploymentTemplateStageConfig},
        {pathToProvisionerTemplate, deploymentTemplateStageProvisioner},
    }
}
```

By using defined `deploymentTemplateStage`s or similar, there is no need to have switch/case based on the path of the templates and things become much more re-usable.